src/dom/element.ts

```diff
@@ -4,9 +4,52 @@ import { Node, NodeType, Text } from "./node.ts";
 import { NodeList, nodeListMutatorSym } from "./node-list.ts";
 
 export class DOMTokenList extends Set<string> {
+  #onChange: (cls: string) => void;
+
+  constructor(onChange: (cls: string) => void) {
+    super();
+    this.#onChange = onChange;
+  }
+
+  add(token: string): this {
+    if (token === "" || /[\t\n\f\r ]/.test(token)) {
+      throw new Error(`DOMTokenList.add: Invalid class token "${token}"`);
+    }
+    super.add(token);
+    this.#onChange([...this].join(" "));
+    return this;
+  }
+
+  clear() {
+    super.clear();
+    this.#onChange("");
+  }
+
+  delete(token: string): boolean {
```

Comment:
While we're at it, `delete` should be a private API with a public `remove` method